.ado/templates/macos-tests-job.yml

```diff
@@ -0,0 +1,36 @@
+parameters:
+  name: ''
+
+jobs:
+  - job: ${{ parameters.name }}
+    displayName: macOS Tests
+    dependsOn: Setup
+    condition: ne( dependencies.Setup.outputs['checkPayload.shouldSkipPRBuild'], 'True' )
+
+    pool:
+      vmImage: $(VmImage)
+    
+    variables:
+      - template: ../variables/macos.yml
```

Comment:
This feels a bit strange where the variables template is used before the pool `$(VmImage)`, 
I know the way ado evaluates things the order doesn't matter especially since it's an template in itself, yet my brain is screaming :D
Perhaps flip these two?